PWGJE/Tasks/PhiInJetsJE.cxx

```diff
@@ -0,0 +1,412 @@
+    // trackSelection = JetDerivedDataUtilities::initialiseTrackSelection(static_cast<std::string>(trackSelections));
+
+  
+  } //end of init
+
+  //  double massKa = TDatabasePDG::Instance()->GetParticle(kKPlus)->Mass();  // FIXME: Get from the common header
+
+  double massKa = o2::analysis::pdg::MassKPlus;
+
+
+  using EventCandidates = soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::MultZeqs>; // , aod::CentFT0Ms, aod::CentFT0As, aod::CentFT0Cs
+  using TrackCandidates = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::TrackSelection,
+				    aod::pidTPCFullKa, aod::pidTOFFullKa>;  
+  Filter jetCuts = aod::jet::pt > c_jetPtMin&& aod::jet::r == nround(c_jetR.node() * 100.0f);
+
+
+  //Function for track quality cuts
+  template <typename TrackType>
+  bool trackSelection(const TrackType track){
+    // basic track cuts
+    if (std::abs(track.pt()) < c_trkMinPt)
+      return false;
+    
+    if (std::abs(track.dcaXY()) > c_MaxDCArToPVcut)
+      return false;
+    
+    if (std::abs(track.dcaZ()) > c_MaxDCAzToPVcut)
+      return false;
+    
+    if (c_PrimaryTrack && !track.isPrimaryTrack())
+      return false;
+    
+    if(std::abs(track.tpcNClsFindable()) < c_nFindableTPCClusters)
+      return false;
+    
+    if(std::abs(track.tpcNClsCrossedRows()) < c_nTPCCrossedRows)
+      return false;
+    
+    if(std::abs(track.tpcCrossedRowsOverFindableCls()) > c_nRowsOverFindable)
+      return false;
+    
+    if(std::abs(track.tpcChi2NCl() > c_nTPCChi2))
+      return false;
+    
+    if(std::abs(track.itsChi2NCl() > c_nITSChi2))
+      return false;
+    
+    if (c_ConnectedToPV && !track.isPVContributor())
+      return false;    
+    return true;
+  };
+
+  template <typename T>
+  bool trackPID(const T& candidate)
+  {
+    bool tpcPIDPassed{false}, tofPIDPassed{false};
+    if (std::abs(candidate.tpcNSigmaKa()) < c_nTPCPID) 
+      tpcPIDPassed = true;
+    
+    if (candidate.hasTOF()) {
+      if (std::abs(candidate.tofNSigmaKa()) < c_nTOFPID) {
+        tofPIDPassed = true;
+      }
+    }
+    
+    else 
+      tofPIDPassed = true;
+    
+    if (tpcPIDPassed && tofPIDPassed) {
+      return true;
+    }
+    return false;
+  }
+
+  
+  template <bool IsMC, bool IsMix, typename TracksType, typename JetType>
+  void MinvReconstruction(double mult, const TracksType& trk1, const TracksType& trk2, const JetType& jets) {
+    TLorentzVector lDecayDaughter1, lDecayDaughter2, lResonance;
+
+    if(!trackSelection(trk1) || !trackSelection(trk2))
+      return;
+      
+    if(!trackPID(trk1) || !trackPID(trk2))
+      return;
+
+    if (trk1.index() == trk2.index())
+      return; // We need to run (0,1), (1,0) pairs as well. but same id pairs are not needed.
+    
+    lDecayDaughter1.SetXYZM(trk1.px(), trk1.py(), trk1.pz(), massKa);
+    lDecayDaughter2.SetXYZM(trk2.px(), trk2.py(), trk2.pz(), massKa);
+    lResonance = lDecayDaughter1 + lDecayDaughter2;
```

Comment:
All this can be done with `RecoDecay`.

Reply:
I prefer to use these for now, as they are familiar to me. I will update them once I can confirm that my code works on hyperloop.